modules/core/map/src/Element/FarmMapInput.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Drupal\farm_map\Element;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\FormElement;
+use Drupal\geofield\GeoPHP\GeoPHPWrapper;
+
+/**
+ * Form element that returns WKT rendered in a map.
+ *
+ * @FormElement("farm_map_input")
+ */
+class FarmMapInput extends FormElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $class = static::class;
+    return [
+      '#input' => TRUE,
+      '#process' => [
+        [$class, 'processElement'],
+      ],
+      '#pre_render' => [
+        [$class, 'preRenderGroup'],
+      ],
+      '#element_validate' => [
+        [$class, 'elementValidate'],
+      ],
+      '#theme_wrappers' => ['form_element'],
+      // Display descriptions above the map by default.
+      '#description_display' => 'before',
+      '#map_type' => 'default',
+      '#map_settings' => [],
+      '#behaviors' => [],
+      '#default_value' => '',
+      '#display_raw_geometry' => FALSE,
+      '#disabled' => FALSE,
+    ];
+  }
+
+  /**
+   * Generates the form element.
+   *
+   * @param array $element
+   *   An associative array containing the properties and children of the
+   *   element. Note that $element must be taken by reference here, so processed
+   *   child elements are taken over into $form_state.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   * @param array $complete_form
+   *   The complete form structure.
+   *
+   * @return array
+   *   The processed element.
+   */
+  public static function processElement(array $element, FormStateInterface $form_state, array &$complete_form) {
+    $element['#tree'] = TRUE;
+
+    // Merge provided map behaviors into defaults. Enable wkt and input
+    // behaviors if #disabled is not TRUE.
+    $default_behaviors = !$element['#disabled'] ? ['wkt', 'input'] : [];
+    $behaviors = array_merge($default_behaviors, $element['#behaviors']);
```

Comment:
This excludes the `wkt` behavior when the element is `#disabled`, but I think we might want to include it? Idea being you want to show the geometry in the map, but not have edit controls. I see we set `behaviors.wkt.edit` when the element is disabled, but that won't do anything if the `wkt` behavior isn't added.

I seem to remember that we add the `wkt` in an event subscriber as well, maybe this is covered there.

Reply:
Yes, the `MapRenderEventSubscriber` still adds the `wkt` behavior whenever WKT is passed in:

https://github.com/mstenta/farmOS/blob/c60b7d159757e8659965fd865b0853ef9e6ec606/modules/core/map/src/EventSubscriber/MapRenderEventSubscriber.php#L61

It is a bit odd to have that logic in two different places now, perhaps. So might be worth bringing that over too in a follow-up commit?